src/utilities/fileSystem.js

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable import/prefer-default-export */
+import RNFS from 'react-native-fs';
+
+const ERRORS = {
+  ERROR_NO_SPACE: { code: 'ERROR_NO_SPACE', message: 'Not enough space on disk to export' },
+  ERROR_NO_FILE: { code: 'ERROR_NO_FILE', message: 'Could not find a realm database' },
+};
+
+/**
+ * Helper method to validate that a file exists and there is enough
+ * space on disk to create a duplicate.
+ * @param  {String} path  Path of the file to check
+ * @return {Object} Object with the shape:
+ * { success: true/false, error:}
+ */
+export async function backupValidation(path) {
+  let returnObject = { success: true };
+  try {
+    // Fetch the size of the realm file.
+    const { size } = await RNFS.stat(path);
+    // Fetch the free space remaining on the device.
+    const { freeSpace } = await RNFS.getFSInfo();
+    // Ensure there is twice the amount of space required for copying the
+    // realm file, to be certain
+    if (freeSpace < size * 2) {
```

Comment:
Is there a reason for needing to ensure there is double the space available? This might be problematic for larger files.

Reply:
Reasoning was that I wanted some buffer than having exactly the number of bytes and `* 2` seemed arbitrarily as good as any other number..!

Reply:
Fair enough, not sure if a constant valued buffer would be better. Having it as a function of the realm file size means that the bigger the file, the bigger the required buffer, which could be problematic?